src/main/java/hec/paireddata/PairedData.java

```diff
@@ -1,27 +1,58 @@
 package hec.paireddata;
 
-import java.sql.Connection;
 import java.util.ArrayList;
+import java.util.function.BiConsumer;
 
 import hec.TimeSeriesDatabase;
 
 public class PairedData {
-    private String sql_table_name;
-    private TimeSeriesDatabase database;
+    private String table_name = null;
+    private TimeSeriesDatabase database = null;
+    private ArrayList<Double> indeps = new ArrayList<>();
+    private ArrayList<Double> deps = new ArrayList<>();
 
-    public PairedData(TimeSeriesDatabase db, String table_name){
+    public PairedData(TimeSeriesDatabase db, String table_name) {
         this.database = db;
+        this.table_name = table_name;    
+    }
 
-        this.sql_table_name = db.sqlTableFor(table_name);
+    public PairedData(String table_name) {
+        this.table_name = table_name;        
     }
 
-    double rate(ArrayList<Double> independant_variables){
-        return rate(independant_variables,1);
+    public void addRow(double indep, double dep) {
```

Comment:
see my comment below, but the notion of fixing this to x,y data only might be short sighted. I think we could have a longer term design discussion about this once we get the basics ironed out.

Reply:
You are correct. I think my comment on PairedDataTest may appropriate to this as well. I did start with multidimensional paired data in mind but decided to focus on the most basic changes since I was doing a relatively large alteration to the TimeSeriesDatabase classes to make room for the concept of other data types than ensemble.

Reply:
Sorry I just realized I lied there. I originally intended to have add row take two array list but I felt it was going to take too long to sort out how to store the data for arbitrary dimensions in a way that made lookups make sense and took the easy path. Perhaps doing to lookup itself should just be left to the other of the object?